### DISCUSSION TITLE
Compiling multiple .cpp file in different folders using Recursive wildcard

### DISCUSSION THREAD

**Discussion by samuelwhiskeyjohnson** [microsoft/vscode-cpptools#8218] [Q&A]
Similar to
https://github.com/microsoft/vscode-cpptools/issues/5502
but I want to use recursive wildcard to search for all .cpp files in all folders in a project file. 

**Folder hierarchy**

+project folder
+----main.cpp
+----add Folder
+--------add.cpp
+--------add.h

**main.cpp**

```
#include <iostream>

using std::cout;

using std::endl;

using std::cin;

#include "addFolder/add.h"



int main() {

int num1 = 50;

int num2 = 100;

int result = addTwoNumbers(num1,num2);

cout << result << endl;return 0;

}

```

**add.h**

```
#ifndef ADDITION_H
#define ADDITION_H
/* header file contents go here */
int addTwoNumbers(int, int);

#endif /* ADDITION_H */
```

**add.cpp**

```
int addTwoNumbers(int inputNum1, int inputNum2) {
return inputNum1 + inputNum2;
}
```

**task.json**
```
{
	"version": "2.0.0",
	"tasks": [
		{
			"type": "cppbuild",
			"label": "C/C++: g++.exe build active file",
			"command": "C:\\raylib\\mingw\\bin\\g++.exe",
			"args": [
				"-g",
				"${workspaceFolder}\\*.cpp",
				"${workspaceFolder}\\addFolder\\*.cpp",
                                //I want to use wildcard instead of providing concrete folder name each time I add new .cpp file in different folder.
				// "${workspaceFolder}\\**\\*.cpp", 
				"-o",
				"${fileDirname}\\${fileBasenameNoExtension}.exe"
			],
			"options": {
				"cwd": "${fileDirname}"
			},
			"problemMatcher": [
				"$gcc"
			],
			"group": {
				"kind": "build",
				"isDefault": true
			},
			"detail": "compiler: C:\\raylib\\mingw\\bin\\g++.exe"
		}
	]
}
```

But "${workspaceFolder}\\**\\*.cpp" doesn't work to compile multiple .cpp files in different folders in project folder. 
Is there anyway how I can compile multiple .cpp files in different folders without providing concrete folder name where .cpp files are at?

**Comment by sean-mcmanus** [comment]
The existing "*" functionality is implemented by gcc, so any "**" passed into the gcc args would also need to be handled by gcc. A potential alternative would be to run some custom build script or building using a build tool such as Make or CMake.

**Reply by samuelwhiskeyjohnson** [reply]
I see. It's not VSCode, it's GCC that doesn't support ** recursive search?

**Reply by sean-mcmanus** [reply]
Yes
